lib/ts/utils.ts

```diff
@@ -207,28 +208,62 @@ export function validateTheStructureOfUserInput(
     }
 }
 
-export async function getAPIVersionFromFileIfExists(): Promise<string | null> {
+export async function getDataFromFileIfExists<T>(filePath: string): Promise<T | undefined> {
```

Comment:
- Why do you need a generic type T? Shouldn't this always return a string | undefined?
- Change the name of the function to `getDataFromFileForServerlessCache`

Reply:
> Why do you need a generic type T? Shouldn't this always return a string | undefined

while storing apiVersion (string) or handshakeInfo (JSON), we JSON.stringify it. So when reading back from the file, when JSON.parse is used, the content of apiVersion file would result in string whereas the content of handshake info would result in JSON.